Code/Tools/AssetProcessor/native/unittests/AssetScannerUnitTests.cpp

```diff
@@ -7,130 +7,145 @@
+        config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder2"), "", "ap3", false, true, platforms));
+        config.AddScanFolder(
+            ScanFolderInfo(tempPath.filePath("subfolder3"), "", "ap4", false, false, platforms)); // note:  "Recurse" set to false.
+        AssetScanner scanner(&config);
+
+        QList<AssetFileInfo> filesFound;
+        QList<AssetFileInfo> foldersFound;
+
+        bool doneScan = false;
+
+        connect(
+            &scanner, &AssetScanner::FilesFound, this,
+            [&filesFound](QSet<AssetFileInfo> fileList)
+            {
+                for (AssetFileInfo foundFile : fileList)
+                {
+                    filesFound.push_back(foundFile);
+                }
+            });
+
+        connect(
+            &scanner, &AssetScanner::FoldersFound, this,
+            [&foldersFound](QSet<AssetFileInfo> folderList)
+            {
+                for (AssetFileInfo foundFolder : folderList)
+                {
+                    foldersFound.push_back(foundFolder);
+                }
+            });
+
+        connect(
+            &scanner, &AssetScanner::AssetScanningStatusChanged, this,
+            [&doneScan](AssetProcessor::AssetScanningStatus status)
+            {
+                if ((status == AssetProcessor::AssetScanningStatus::Completed) || (status == AssetProcessor::AssetScanningStatus::Stopped))
+                {
+                    doneScan = true;
+                }
+            });
+
+        // this test makes sure that no files that should be missed were actually missed.
+        // it makes sure that if a folder is added recursively, child files and folders are found
+        // it makes sure that if a folder is added NON-recursively, child folder files are not found.
+
+        scanner.StartScan();
+        QElapsedTimer nowTime;
+        nowTime.start();
+        while (!doneScan)
         {
-            foldersFound.push_back(foundFolder);
-        }
-    }
-    );
+            QCoreApplication::processEvents(QEventLoop::WaitForMoreEvents, 100);
 
-    connect(&scanner, &AssetScanner::AssetScanningStatusChanged, this, [&doneScan](AssetProcessor::AssetScanningStatus status)
-        {
-            if ((status == AssetProcessor::AssetScanningStatus::Completed) || (status == AssetProcessor::AssetScanningStatus::Stopped))
+            if (nowTime.elapsed() > 10000)
             {
-                doneScan = true;
+                break;
             }
         }
-        );
 
-    // this test makes sure that no files that should be missed were actually missed.
-    // it makes sure that if a folder is added recursively, child files and folders are found
-    // it makes sure that if a folder is added NON-recursively, child folder files are not found.
+        EXPECT_TRUE(doneScan);
+        EXPECT_TRUE(filesFound.count() == expectedFiles.count());
```

Comment:
Since you already have to change these lines, it might be a good idea to look for opportunities to switch to better EXPECT_ macros like EXPECT_EQ in this case

Reply:
I agree .
`EXPECT_TRUE` should only be used of a statement that returns a boolean.

When operation is using a relational operator such as ==, >, !=, etc... it should use the appropriate EXPECT_ macros.

Reply:
I've updated the tests to use better EXPECTs